test/jdk/java/net/UnixDomainSocketAddress/LengthTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Test UnixDomainSocketAddress constructor
+ * @library /test/lib
+ * @run testng/othervm LengthTest
+ */
+
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static java.lang.System.out;
+import static java.net.StandardProtocolFamily.UNIX;
+import static jdk.test.lib.Asserts.assertTrue;
+
+import java.net.UnixDomainSocketAddress;
+import java.io.IOException;
+import java.nio.channels.SocketChannel;
+import java.nio.file.Path;
+
+public class LengthTest {
+    static boolean supported = true;
+    final int namelen = 100;    // length close to max
+
+    @BeforeTest
+    public void setup() {
+        try {
+            SocketChannel.open(UNIX);
+        }
+        catch (IOException | UnsupportedOperationException e) {
+            supported = false;
+            out.println("Unix domain channels not supported. Test not run.");
```

Comment:
`SkipException` can be used to avoid the boolean and the subsequent checks in all test scenario methods. e.g.

```
throw new org.testng.SkipException("Unix domain channels not supported");
```

This pattern can be used in all tests at are optional - based on platform support.

Additionally, if UOE is thrown from SocketChannel.open(UNIX), then I assume that ServerSocketChannel.open(UNIX) should also throw UOE. This can be asserted in the catch.

Reply:
Actually, tests of the address class should pass on all platforms. The check dated back to when the test needed to know the exact maximum length of an address on the platform. I'll just remove the check.